packages/blob/src/index.ts

```diff
@@ -215,13 +231,152 @@ export async function list(
   };
 }
 
-function getApiUrl(): string {
-  return process.env.VERCEL_BLOB_API_URL || 'https://blob.vercel-storage.com';
+export interface GenerateClientTokenOptions extends BlobCommandOptions {
+  pathname: string;
+  onUploadCompleted?: {
+    callbackUrl: string;
+    metadata?: string;
+  };
+  maximumSizeInBytes?: number;
+  allowedContentTypes?: string[];
+}
+
+export async function generateClientTokenFromReadWriteToken({
+  token,
+  ...args
+}: GenerateClientTokenOptions): Promise<string> {
+  if (typeof window !== 'undefined') {
+    throw new Error(
+      '"generateClientTokenFromReadWriteToken" must be called from a server environment',
+    );
+  }
+  const timestamp = new Date();
+  timestamp.setSeconds(timestamp.getSeconds() + 30);
+  const blobToken = getToken({ token });
+
+  const [, , , storeId = null] = blobToken.split('_');
+
+  if (!storeId) {
+    throw new Error(
+      token ? 'Invalid "token" parameter' : 'Invalid BLOB_READ_WRITE_TOKEN',
```

Comment:
nice 👍 I am not sure we do so for other methods accepting token or BLOB_READ_WRITE_TOKEN

Reply:
it's not needed in the other methods, as `option.token` falls back to `BLOB_READ_WRITE_TOKEN`

Reply:
oh right 👍

Reply:
we only parse the token in this method. In the other cases we return the message `Vercel Blob: Access denied, please provide a valid token for this resource` in case of an unauthorized response, or we throw `BLOB_READ_WRITE_TOKEN environment variable is not set. Please set it to your write token` in case there are no tokens defined.